### PR TITLE
[SIMT] Add all_sync warp intrinsics

### PR DESCRIPTION
**Pull request by varinic**
Related issue = #4631
Add all_sync intrinsics for cuda backend.

### CODE REVIEW

**Comment by qiao-bo** [comment]
Hi, thanks for contributing, could you please resolve the conflict first ;/?

File: taichi/runtime/llvm/runtime.cpp [hunk 1]
```diff
@@ -1036,6 +1036,18 @@ f32 cuda_shfl_up_sync_f32(u32 mask, f32 val, i32 delta, int width) {
   return 0;
 }
 
+bool cuda_all_sync(u32 mask, bool bit) {
+  return false;
+}
+
+int32 cuda_all_i32(int32 predicate) {
```
Comment (YuCrazing) [inline taichi/runtime/llvm/runtime.cpp:1043]: Same question: Is this function redundant? It seems that `cuda_all_i32` can be replaced by `cuda_all_sync_i32`. If you want to keep this function, please expose it to python and add corresponding unit tests. :)

**Review by YuCrazing — CHANGES_REQUESTED** [review]

File: taichi/runtime/llvm/runtime.cpp (hunk 1 above)
Comment (varinic) [reply taichi/runtime/llvm/runtime.cpp]: I will remove `cuda_all_i32`.

**Review by YuCrazing — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**